File: e2e/playwright-typescript/interactions/validators/TaskValidator.ts
import { Page, expect } from '@playwright/test';
import { TaskInfoModal } from '@/interactions/pages/TaskInfoModal';
import { TaskData } from '@/data/TaskData';
import { step } from '@/decorators/step-decorator';

export class TaskValidator {
  private readonly modal: TaskInfoModal;

  constructor(page: Page) {
    this.modal = new TaskInfoModal(page);
  }

  @step('Validate task info data: {taskData}')
  async data(task: TaskData): Promise<this> {
    await expect(this.modal.title).toHaveText(task.title);
    await expect(this.modal.description).toHaveText(task.description);
    await expect(this.modal.statusTag(task.status)).toBeVisible();
    await expect(this.modal.priorityTag(task.priority)).toBeVisible();
    return this;
  }

  @step('Validate task is marked as valid')
  async isValid(): Promise<this> {
    await expect(this.modal.validationBadge).toBeVisible();
    return this;
  }

  @step('Validate task is marked as not valid')
  async isNotValid(): Promise<this> {
    await expect(this.modal.validationBadge).not.toBeVisible();
    return this;
  }
}

File: e2e/playwright-python/interactions/validators/task_validator.py
from playwright.sync_api import Page, expect
from data.task_data import TaskData
from interactions.pages.task_info_modal import TaskInfoModal
from decorators.step_decorator import step


class TaskValidator:
    def __init__(self, page: Page):
        self._modal = TaskInfoModal(page)

    @step("Validate task info data: {task_data}")
    def data(self, task_data: TaskData) -> "TaskValidator":
        expect(self._modal.title).to_have_text(task_data.title)
        expect(self._modal.description).to_have_text(task_data.description)
        expect(self._modal.status_tag(task_data.status)).to_be_visible()
        expect(self._modal.priority_tag(task_data.priority)).to_be_visible()
        return self

    @step("Validate task is marked as valid")
    def is_valid(self) -> "TaskValidator":
        expect(self._modal.validation_badge).to_be_visible()
        return self

    @step("Validate task is marked as not valid")
    def is_not_valid(self) -> "TaskValidator":
        expect(self._modal.validation_badge).not_to_be_visible()
        return self
